bundle/verify.go

```diff
@@ -20,9 +20,22 @@ import (
 	"github.com/pkg/errors"
 )
 
+const defaultVerifierID = "_default"
+
+var verifiers map[string]Verifier
+
+// Verifier is the interface expected for implementations that verify bundle signatures.
+type Verifier interface {
+	VerifyBundleSignature(SignaturesConfig, *VerificationConfig) (map[string]FileInfo, error)
+}
+
+// DefaultVerifier is the default bundle verification implementation. It verifies bundles by checking
+// the JWT signature using a locally-accessible public key.
+type DefaultVerifier struct{}
+
 // VerifyBundleSignature verifies the bundle signature using the given public keys or secret.
 // If a signature is verified, it keeps track of the files specified in the JWT payload
-func VerifyBundleSignature(sc SignaturesConfig, bvc *VerificationConfig) (map[string]FileInfo, error) {
+func (*DefaultVerifier) VerifyBundleSignature(sc SignaturesConfig, bvc *VerificationConfig) (map[string]FileInfo, error) {
```

Comment:
Won't this break existing clients integrating OPA after an update ? Same for the `GenerateSignedToken`.  We probably need something like 
```go
func GenerateSignedToken(files []FileInfo, sc *SigningConfig, keyID string) (string, error) {
     // get the default signer
     // call the default signer's generate method
}
```

Reply:
hmm good point, let me take another look at this.

Reply:
ok, brought back the original interface, lmk if that looks ok